Code clean up and documentation

rickardlindberg opened the pull request.

rickardlindberg committed:
Problem: Unnecessary glue module Language.

The DSL module can now be exposed publicly since all private
parts can remain private.

The Uno and Library modules only need to use the public API
of the DSL. This was not the case before.

rickardlindberg committed:
Improve FRP intro text.

rickardlindberg committed:
Fix typo.

rickardlindberg committed:
Improve FRP transforming text.

rickardlindberg committed:
Explain folding.

rickardlindberg committed:
Explain filtering.

mvcisback commented on README.md:
```diff
@@ -39,34 +41,75 @@ Lets explore them in more detail.
+
+![Counting number of clicks.](doc/stream-fold.png)
+
+The very firs time `clickCount` is 0. Subsequent values are incremented by one
```
first*

mvcisback commented on doc/frp.md:
```diff
@@ -1,29 +1,70 @@
+
+![Counting number of clicks.](doc/stream-fold.png)
+
+The very firs time `clickCount` is 0. Subsequent values are incremented by one
```
again: firs :arrow_right:  first

rickardlindberg committed:
Fix spell error.

rickardlindberg committed:
Problem: Multiple commands are needed to verify a patch.

I prefer a single command to check if a patch is good. Since
travis now checks spelling of README, that should also be included
in the test command.

rickardlindberg commented:
According to C4, I think you should have merged this pull request before the spell error was fixed:

> Maintainers SHALL merge correct patches from other Contributors rapidly.

The question then is if my patch was "correct". C4 defines "correct patch" under "Patch requirements". From what I understand, it is a correct patch.

This is what I think happened this time:

> Maintainers SHALL NOT make value judgments on correct patches.

I agree that the spell error should be fixed. But it should be fixed by making a new pull request.

I'm trying to understand and apply C4. Does this make sense to you?

rickardlindberg commented:
I'm not sure how to work with pull requests that contains different sets of changes. Perhaps I should have created multiple branches on my end and send different pull requests.

This pull request now also includes moving spell check command from travis file to test script.

mvcisback commented:
I think that's fine for this patch. In the future multiple branches is ideal

rickardlindberg commented:
Regarding multiple branches: I see a problem with that if you have commits that depend on each other. For example:

```
master
     \----> branch1
                  \-------> branch2
```

If I first push branch1 and create a pull request, then work on another change in branch2, and that change depends on the code in branch1. If I push branch2 and create a pull request, that pull request will include all changes from branch1 as well. So it is equivalent to continue to do the changes in branch1 and then push that one again so the pull request will be updated with the new commits.
